### PR TITLE
Pass exactly the same string to vstest.console which user has passed to `dotnet test`

### PR DESCRIPTION
**Pull request by Faizan2304**
E2E flow of `dotnet test` is divided into three layer.
1) user call `dotnet test`
2) `dotnet test` call vstest target through msbuild with argument which user has passed to `dotnet test`
3) vstest task call vstest.console with the argument which msbuild has passed to vstest task.

Between this layer arguments lost their original form.

For example:
Suppose user ran the following command
`dotnet test -r ".\result directory\\"`

## Expected
vstest task should call vstest.console.exe as follows
`dotnet vstest.console.dll --resultDirectory:".\result directory\\" test.dll`

## Actual
vstest task is calling vstest.console.exe as follows
`dotnet vstest.console.dll --resultDirectory:".\result directory\" test.dll`

As a result of which `".\result directory\" test.dll` is getting read as one argument.

issue:#878

### CODE REVIEW

File: src/Microsoft.TestPlatform.Build/ArgumentEscaper.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        /// <summary>
+        /// Undo the processing which took place to create string[] args in Main,
+        /// so that the next process will receive the same string[] args
+        /// 
+        /// See here for more info:
+        /// http://blogs.msdn.com/b/twistylittlepassagesallalike/archive/2011/04/23/everyone-quotes-arguments-the-wrong-way.aspx
+        /// </summary>
+        /// <param name="arg"></param>
+        /// <returns>Return original string passed by client</returns>
+        public static string EscapeArgForProcessStart(string arg)
```
Comment (codito) [inline src/Microsoft.TestPlatform.Build/ArgumentEscaper.cs]: nit: not sure what the name signifies

File: src/Microsoft.TestPlatform.Build/ArgumentEscaper.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        {
+            var sb = new StringBuilder();
+
+            var needsQuotes = ShouldSurroundWithQuotes(arg);
+            var isQuoted = needsQuotes || IsSurroundedWithQuotes(arg);
+
+            if (needsQuotes) sb.Append("\"");
```
Comment (codito) [inline src/Microsoft.TestPlatform.Build/ArgumentEscaper.cs]: nit: parenthesis for if clause

File: src/Microsoft.TestPlatform.Build/ArgumentEscaper.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+                // Output any consumed backslashes and the character
+                else
+                {
+                    sb.Append('\\', backslashCount);
+                    sb.Append(arg[i]);
+                }
+            }
+
+            if (needsQuotes) sb.Append("\"");
```
Comment (codito) [inline src/Microsoft.TestPlatform.Build/ArgumentEscaper.cs]: nit: parenthesis

File: test/Microsoft.TestPlatform.Build.UnitTests/ArgumentEscaperTests.cs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+namespace Microsoft.TestPlatform.Build.Utils.UnitTests
+{
+    using Microsoft.TestPlatform.Build.Utils;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    [TestClass]
+    public class ArgumentEscaperTests
+    {
+        [TestMethod]
+        public void EscapeArgForProcessStartShouldAddDoubleQuoteIfThereIsSpace()
```
Comment (codito) [inline test/Microsoft.TestPlatform.Build.UnitTests/ArgumentEscaperTests.cs]: Do we need tests for single quote, forward slash, space at end?

**Review by codito — APPROVED** [review]

File: src/Microsoft.TestPlatform.Build/ArgumentEscaper.cs (hunk 1 above)
Comment (Faizan2304) [inline src/Microsoft.TestPlatform.Build/ArgumentEscaper.cs]: I renamed it to HandleEscapeSequenceInArgForProcessStart. This method make changes in args to pass Process.Start. We we have to pass this to cmd then args will get handle in some other way.

File: src/Microsoft.TestPlatform.Build/ArgumentEscaper.cs (hunk 2 above)
Comment (Faizan2304) [inline src/Microsoft.TestPlatform.Build/ArgumentEscaper.cs]: Fixed

File: src/Microsoft.TestPlatform.Build/ArgumentEscaper.cs (hunk 3 above)
Comment (Faizan2304) [inline src/Microsoft.TestPlatform.Build/ArgumentEscaper.cs]: Fixed

File: test/Microsoft.TestPlatform.Build.UnitTests/ArgumentEscaperTests.cs (hunk 4 above)
Comment (Faizan2304) [inline test/Microsoft.TestPlatform.Build.UnitTests/ArgumentEscaperTests.cs]: Fixed

**Pull request merged**